test/test.oauth2.ts

```diff
@@ -46,14 +47,17 @@ describe('OAuth2 client', () => {
         new OAuth2Client(CLIENT_ID, CLIENT_SECRET, REDIRECT_URI);
     const generated = oauth2client.generateAuthUrl(opts);
     const parsed = url.parse(generated);
-    const query = qs.parse(parsed.query);
-
-    assert.equal(query.response_type, 'code token');
-    assert.equal(query.access_type, ACCESS_TYPE);
-    assert.equal(query.scope, SCOPE);
-    assert.equal(query.client_id, CLIENT_ID);
-    assert.equal(query.redirect_uri, REDIRECT_URI);
-    done();
+    if (typeof parsed.query === 'string') {
```

Comment:
nit: reverse this branch. It is preferable to write branches with error condition throw first. Errors return early and so neither the indent not the early return for the main logic would be necessary.

Reply:
Yikes, weird I missed that.  Fixed.